pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -553,15 +497,37 @@ func (c *ConnectedObjectsQuery) reverseExpandDirect(
 		}
 
 		subg.Go(func() error {
-			return c.execute(subgctx, &ConnectedObjectsRequest{
+			return c.Execute(subgctx, &ConnectedObjectsRequest{
+				Ingress:          req.ingress,
 				StoreID:          store,
 				ObjectType:       targetObjectType,
 				Relation:         targetObjectRel,
 				User:             sourceUserRef,
 				ContextualTuples: req.contextualTuples,
-			}, resultChan, foundObjectsMap, foundCount)
+			}, resultChan)
 		})
 	}
 
 	return subg.Wait()
 }
+
+func (c *ConnectedObjectsQuery) sendCandidate(ctx context.Context, ingress *graph.RelationshipIngress, candidateObject string, candidateChan chan<- *ConnectedObjectsResult) {
+	if _, ok := c.candidateObjectsMap.LoadOrStore(candidateObject, struct{}{}); !ok {
+		if c.candidatesFound != nil && atomic.AddUint32(c.candidatesFound, 1) > c.maxCandidates {
+			return
+		}
+		_, span := tracer.Start(ctx, "sendCandidate", trace.WithAttributes(
```

Comment:
Should we start this at the beginning of the function call. It's possible with enough contention on the `LoadOrStore` that we could see higher trace latencies for this function, but with where this is placed we won't ever see contention on the synchronized map.

Reply:
I put this trace here so that when I was debugging via looking at Jaeger, I could quickly see which objects were sent through the channel. I didn't care about performance. If I move the trace outside the first `if`, then `sendCandidate` is not an accurate name for the trace...

Reply:
See the original PR description with a screenshot of how tracing looks like

Reply:
> This pattern/usage of a span is odd and feels very much non-standard. Instead, can we consider starting the span at the top level so that it is still representative of any latency incurred in this function call but also add an attribute indicating if a result was yielded

ok